src/collector.ts

```diff
@@ -43,28 +45,32 @@ export const collector = async ({
         'session_recorders',
         'third_party_trackers'
     ],
-    puppeteerExecutablePath = null,
-    extraChromiumArgs = []
-}) => {
-    clearDir(outDir);
+    puppeteerExecutablePath: null as string|null,
+    extraChromiumArgs: [] as string[],
+    extraPuppeteerOptions: {} as Partial<PuppeteerLaunchOptions>
+}
+
+export const collector = async (inUrl: string, args: CollectorOptions) => {
+    args = { ...DEFAULT_OPTIONS, ...args };
+    clearDir(args.outDir);
     const FIRST_PARTY = parse(inUrl);
     let REDIRECTED_FIRST_PARTY = parse(inUrl);
-    const logger = getLogger({ outDir, quiet });
+    const logger = getLogger({ outDir: args.outDir, quiet: args.quiet });
 
     const output: any = {
-        title,
+        args: args.title,
         uri_ins: inUrl,
         uri_dest: null,
         uri_redirects: null,
         secure_connection: {},
         host: url.parse(inUrl).hostname,
         config: {
-            clearCache,
-            captureHar,
-            captureLinks,
-            enableAdBlock,
-            emulateDevice,
-            numPages
+            cleareCache: args.clearCache,
+            captureHar: args.captureHar,
+            captureLinks: args.captureLinks,
+            enableAdBlock: args.enableAdBlock,
+            emulateDevice: args.emulateDevice,
+            numPages: args.numPages
```

Comment:
Might as well just pass the entire configuration object. Though, it's sort of odd to do that since the user has the configuration. Again, I'm just trying to think about how it might be more useful as a package.

Reply:
Yeah I think returning this is unnecessary, but we can think more about what it should do here in a different PR.